### ISSUE TITLE
DSDL Compiler

### ISSUE THREAD

**Issue by flybrianfly** [dronecan/libuavcan#3]
Hello,

Embedded C/C++ developer here, hoping to integrate Drone CAN with my projects. Sorry in advance that I'm a novice with python. I'm on Ubuntu 20.04 with build-essential and CMake installed. Cloned this repo, updated the git submodules, created a build directory, ran cmake .., and am getting the following error output, which I'm also attaching a screenshot in case that is easier to read. Thanks in advance for any clues!

```
[  1%] Running dsdl compiler
Traceback (most recent call last):
  File "/home/brian/Documents/Software/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdlc", line 59, in <module>
    from libuavcan_dsdl_compiler import run as dsdlc_run
  File "/home/brian/Documents/Software/libuavcan/libuavcan/dsdl_compiler/libuavcan_dsdl_compiler/__init__.py", line 18, in <module>
    from dronecan import dsdl
  File "/home/brian/Documents/Software/libuavcan/libuavcan/dsdl_compiler/pyuavcan/dronecan/__init__.py", line 216, in <module>
    import dronecan.app as app
  File "/home/brian/Documents/Software/libuavcan/libuavcan/dsdl_compiler/pyuavcan/dronecan/app/__init__.py", line 12, in <module>
    from . import dynamic_node_id
  File "/home/brian/Documents/Software/libuavcan/libuavcan/dsdl_compiler/pyuavcan/dronecan/app/dynamic_node_id.py", line 25, in <module>
    class CentralizedServer(object):
  File "/home/brian/Documents/Software/libuavcan/libuavcan/dsdl_compiler/pyuavcan/dronecan/app/dynamic_node_id.py", line 26, in CentralizedServer
    QUERY_TIMEOUT = uavcan.protocol.dynamic_node_id.Allocation().FOLLOWUP_
![Screenshot from 2022-01-06 17-40-25](https://user-images.githubusercontent.com/7801107/148472617-f469c051-4ba1-48d8-b7d3-78703966ada9.png)
TIMEOUT_MS / 1000  # @UndefinedVariable
AttributeError: 'Namespace' object has no attribute 'protocol'
make[2]: *** [libuavcan/CMakeFiles/libuavcan_dsdlc.dir/build.make:172: libuavcan_dsdlc_run.stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:372: libuavcan/CMakeFiles/libuavcan_dsdlc.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

![Screenshot from 2022-01-06 17-40-25](https://user-images.githubusercontent.com/7801107/148472642-0317396b-fbab-4830-aed8-9b0a6d1cbc36.png)

**Comment by flybrianfly** [comment]
> I hit the same issue when updating the libuavcan submodule in PX4, but haven't had a chance to look into it yet. As a temporary workaround you could start a few commits back at [9c09983](https://github.com/dronecan/libuavcan/commit/9c09983f737de0dd224d26025f8f439d6d860b27).
> 
> FYI @tridge

Sure enough, using [9c09983](https://github.com/dronecan/libuavcan/commit/9c09983f737de0dd224d26025f8f439d6d860b27) fixed the issue for me. Thanks!

**Comment by marcojob** [comment]
Have the same issue using master. Bumping the PX4 submodule to master of libuavcan will be required eventually as the update of pyuavcan in #2 is required for python 3.10 which ships with Ubuntu 22.04 by default. So more people will run into the issue soon-ish. Maybe I find the time to fix this

**Comment by marcojob** [comment]
The PR mentioned above (dronecan/pydronecan#21) should fix this issue. If everyone here agrees with the fix we could then update PX4 to latest master of this repository

**Comment by phamelin** [comment]
@marcojob I have the same issue. Is there a plan to merge the fix yet? Thank you!

**Comment by marcojob** [comment]
> @marcojob I have the same issue. Is there a plan to merge the fix yet? Thank you!

I am not sure - the maintainers of this repo have to decide what the plan is

**Comment by julianoes** [comment]
Fixed in #7 from what I can tell, can be closed.
